### PR TITLE
Split `stableBufferTime` into `hybridSwitchBufferTime` and `stableBufferTime`

### PR DESCRIPTION
**Pull request by eirikbjornr**
## What?

Split `stableBufferTime` into `hybridSwitchBufferTime` and `stableBufferTime` so the threshold to switch between BOLA and throughput ABR algorithms (`hybridSwitchBufferTime`) can be configured separately from `stableBufferTime`.

> Ticket: [BADGERS-214](https://jira.dev.bbc.co.uk/browse/BADGERS-214)

### CODE REVIEW

File: test/unit/streaming.models.MediaPlayerModel.js [hunk 1]
```diff
@@ -302,6 +302,22 @@ describe('MediaPlayerModel', function () {
         expect(stableBufferTime).to.equal(10);
     });
 
+    it('should configure HybridSwitchBufferTime', function() {
```
Comment (matt-stephenson) [inline test/unit/streaming.models.MediaPlayerModel.js:305]: Minor, but could this include the stableBufferTime too, just to ensure that the OR guard is the right way round if both exist?

**Review by matt-stephenson — APPROVED** [review]
Looks good. Minor comment, but not blocking.

File: test/unit/streaming.models.MediaPlayerModel.js (hunk 1 above)
Comment (eirikbjornr) [reply test/unit/streaming.models.MediaPlayerModel.js:305]: Good shout!

File: src/core/Settings.js [hunk 2]
```diff
@@ -863,6 +866,7 @@ function Settings() {
                 bufferTimeAtTopQualityLongForm: 60,
                 initialBufferLevel: NaN,
                 stableBufferTime: 12,
+                hybridSwitchBufferTime: NaN,
```
Comment (jlks) [inline src/core/Settings.js:869]: Should this default to 12 given the comment above?

Comment (eirikbjornr) [reply src/core/Settings.js:869]: Good spot! Forgot to change the default value to be `stableBufferTime` in documentation. Is it clearer with the change?

File: test/unit/streaming.models.MediaPlayerModel.js (hunk 1 above)
Comment (eirikbjornr) [reply test/unit/streaming.models.MediaPlayerModel.js:305]: Done

**Pull request merged**